sql-plugin/src/main/scala/org/apache/spark/sql/rapids/AggregateFunctions.scala

```diff
@@ -76,20 +76,17 @@ case class GpuAggregateExpression(origAggregateFunction: GpuAggregateFunction,
   case class WrappedAggFunction(aggregateFunction: GpuAggregateFunction, filter: Expression)
     extends GpuDeclarativeAggregate {
     override val inputProjection: Seq[GpuExpression] = {
-      val caseWhenExpressions = aggregateFunction.inputProjection.map {ip =>
+      val caseWhenExpressions = aggregateFunction.inputProjection.map { ip =>
         // special case average with null result from the filter as expected values should be
         // (0.0,0) for (sum, count)
-        val initialValue : Expression =
-          if (origAggregateFunction.prettyName.contains("gpuavg")) {
-            if (ip.dataType == DoubleType) {
-              GpuLiteral(0D, DoubleType)
-            } else {
-              GpuLiteral(0L, LongType)
+        val initialValue: Expression =
+          origAggregateFunction.asInstanceOf[GpuDeclarativeAggregate] match {
```

Comment:
why do we need the `asInstanceOf` here? Just curious. Maybe we should blow up if `origAggregateFunction` is not a `GpuDeclarativeAggregate`, but I think it'll still match without it being a declarative agg.

Reply:
my bad, debugging artifact. Removed now.